hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/CryptoUtils.java

```diff
@@ -55,15 +58,18 @@ public static String getJceProvider(Configuration conf) {
         CommonConfigurationKeysPublic.HADOOP_SECURITY_CRYPTO_JCE_PROVIDER_AUTO_ADD_KEY,
         CommonConfigurationKeysPublic.HADOOP_SECURITY_CRYPTO_JCE_PROVIDER_AUTO_ADD_DEFAULT);
 
-    // For backward compatible, auto-add BOUNCY_CASTLE_PROVIDER_CLASS.
-    if (autoAdd && !provider.isEmpty()) {
+    // For backward compatible, auto-add BOUNCY_CASTLE_PROVIDER_CLASS when the provider is "BC".
+    if (autoAdd && PROVIDER_NAME.equals(provider)) {
       try {
         // Use reflection in order to avoid statically loading the class.
         final Class<?> clazz = Class.forName(BOUNCY_CASTLE_PROVIDER_CLASS);
-        final Field provider_name = clazz.getField("PROVIDER_NAME");
-        if (provider.equals(provider_name.get(null))) {
+        final Field providerName = clazz.getField("PROVIDER_NAME");
```

Comment:
I dont think this is needed any more. If it is, use the constant `PROVIDER_NAME_FIELD`, but really, given we know what string we are looking for, no need to ask for the field or check it again